### ISSUE TITLE
'can-i-deploy --latest' no longer works on latest version of cli

### ISSUE THREAD

**Issue by bfoley19** [pact-foundation/pact-ruby-cli#12]
Same exact command on version 0.1.0-1 which succeeds will fail on version 0.2.0-1.

`docker run --rm -e PACT_BROKER_BASE_URL=***** -e PACT_BROKER_TOKEN=****** pactfoundation/pact-cli:0.1.0-1 broker can-i-deploy --pacticipant someConsumer --latest --retry-while-unknown=10 --retry-interval=5`

`someConsumer | d02ac69214f19d2d763155164dc... | someProvider | cfd661a3803e531af8dbdaa0ceb... | true`

Same command on version 0.2.0-1 fails after retry interval is done with:
`someConsumer | d02ac69214f19d2d763155164dc... | someProvider | ???       | ???`

**Comment by bethesque** [comment]
Can you turn verbose mode on for each of them and copy paste the output?

**Comment by bethesque** [comment]
Also, if you can post the relevant matrix rows so I can understand the underlying data, that would help.

**Comment by bethesque** [comment]
Also, can you tell me, when you go to `https://your-broker/pacticipants/someProvider/latest-version`, is `cfd661a3803e531af8dbdaa0ceb` the latest version?

**Comment by bfoley19** [comment]
@bethesque Dang, I'm having some issues reproducing the same issue now.  I had it in a state where i could swap between commands using version 2* and version 1* and both were consistently contradicting each other on verified true and verified ??? (for 2*).  I wish I would have captured the verbose output at the time.

When I get some time I might play with the versions for a bit to see if I can get a deterministic repo case for you.   If you prefer closing this until then I can open it if/when I run into this again.

**Comment by bethesque** [comment]
I think I know what it might be.

https://github.com/pact-foundation/pact_broker-client/blob/master/CHANGELOG.md#bug-fixes-1

I realised that the code was out of sync with the documentation, and fixed the code. The docs said "if you specify one pacticipant version and you don't specify a `--to ENV` it will calculate whether the app is compatible with the *latest* version of each of the apps with which it integrates". When I looked at the code, I realised that it was checking *all* the rows of the apps with which it integrates, not the latest. My guess is that the earlier version of the cli was incorrectly reporting a success when it should have been a fail. The successful result it was getting might not have been the result for the *latest* version of someProvider.
